File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Reflection;
using DevExpress.XtraEditors;
using DevExpress.XtraBars.Docking2010.Views.Tabbed;
using DevExpress.XtraBars.Docking2010;
using DevExpress.XtraBars.Docking;
using DevExpress.XtraRichEdit;
using DevExpress.XtraBars.Docking2010.Views;

namespace ChildStandaloneForm
{
    public partial class Form1 : XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            tabbedView1.FloatingDocumentContainer = DevExpress.XtraBars.Docking2010.Views.FloatingDocumentContainer.DocumentsHost;
        }

        List<BaseDocument> docs = new List<BaseDocument>();
        private void tabbedView1_QueryControl(object sender, DevExpress.XtraBars.Docking2010.Views.QueryControlEventArgs e)
        {
            MyBrowser wb = new MyBrowser();
            wb.Dock = DockStyle.Fill;
            wb.TitleChanged += wb_TitleChanged;
            docs.Add(e.Document);
            e.Control = wb;
        }

        private Document FindDocument(object documentControl)
        {
            foreach (Document doc in docs)
                if (doc.Control == documentControl)
                    return doc;
            return null;
        }

        void wb_TitleChanged(object sender, EventArgs e)
        {
            tabbedView1.ActiveDocument.Caption = (tabbedView1.ActiveDocument.Control as MyBrowser).Title;
            Document activeFloatDoc = FindDocument(sender); // for floating documents
            if (activeFloatDoc != null)
                activeFloatDoc.Caption = (sender as MyBrowser).Title;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraBars.Docking2010.Views.Tabbed
Imports DevExpress.XtraBars.Docking2010.Views

Namespace ChildStandaloneForm

    Public Partial Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            tabbedView1.FloatingDocumentContainer = FloatingDocumentContainer.DocumentsHost
        End Sub

        Private docs As List(Of BaseDocument) = New List(Of BaseDocument)()

        Private Sub tabbedView1_QueryControl(ByVal sender As Object, ByVal e As QueryControlEventArgs)
            Dim wb As MyBrowser = New MyBrowser()
            wb.Dock = DockStyle.Fill
            AddHandler wb.TitleChanged, AddressOf wb_TitleChanged
            docs.Add(e.Document)
            e.Control = wb
        End Sub

        Private Function FindDocument(ByVal documentControl As Object) As Document
            For Each doc As Document In docs
                If doc.Control Is documentControl Then Return doc
            Next

            Return Nothing
        End Function

        Private Sub wb_TitleChanged(ByVal sender As Object, ByVal e As EventArgs)
            tabbedView1.ActiveDocument.Caption = TryCast(tabbedView1.ActiveDocument.Control, MyBrowser).Title
            Dim activeFloatDoc As Document = FindDocument(sender) ' for floating documents
            If activeFloatDoc IsNot Nothing Then activeFloatDoc.Caption = TryCast(sender, MyBrowser).Title
        End Sub
    End Class
End Namespace
